File: src/main/java/META-INF/resources/dataforms/validator/DateValidator.js
/**
 * @fileOverview {@link DateValidator}クラスを記述したファイルです。
 */

'use strict';

import { DateTimeValidator } from './DateTimeValidator.js';

/**
 * @class DateValidator
 * 日付バリデータクラス。
 * <pre>
 * </pre>
 * @extends DateTimeValidator
 */
export class DateValidator extends DateTimeValidator {
	/**
	 * コンストラクタ。
	 */
	constructor() {
		super();
		this.dateFormatKey = "format.datefield";
	}
}

File: src/main/java/jp/dataforms/fw/validator/DateValidator.java
package jp.dataforms.fw.validator;

import jp.dataforms.fw.util.MessagesUtil;


/**
 * 日付バリデータクラス。
 *
 */
public class  DateValidator extends DateTimeValidator {
	/**
	 * コンストラクタ。
	 */
	public DateValidator() {
		super("error.date");
	}

	@Override
	public boolean validate(final Object value) {
		String format = MessagesUtil.getMessage(this.getPage(), "format.datefield");
		this.setDateFormat(format);
		return super.validate(value);
	}

}

